### ISSUE TITLE
Error in GitHubReleasesInfoProvider with 2.8.1RC6

### ISSUE THREAD

**Issue by vmiller** [autopkg/autopkg#877]
**THIS IS ONLY INTENDED FOR AUTOPKG BETAS.**

**Describe the problem**
When testing recipes against 2.8.1 RC6 a number are failing with an error in GitHubReleasesInfoProvider.  Traceback : 

```
Traceback (most recent call last):
  File "/Library/AutoPkg/autopkglib/__init__.py", line 873, in process
    self.env = processor.process()
  File "/Library/AutoPkg/autopkglib/__init__.py", line 656, in process
    self.main()
  File "/Library/AutoPkg/autopkglib/GitHubReleasesInfoProvider.py", line 202, in main
    releases = self.get_releases(
  File "/Library/AutoPkg/autopkglib/GitHubReleasesInfoProvider.py", line 132, in get_releases
    github = autopkglib.github.GitHubSession(
  File "/Library/AutoPkg/autopkglib/github/__init__.py", line 56, in __init__
    self.token = self._get_token(token_path=token_abspath)
  File "/Library/AutoPkg/autopkglib/github/__init__.py", line 64, in _get_token
    if not os.path.exists(pref_token_path) and os.path.exists(token_path):
  File "/Library/AutoPkg/Python3/Python.framework/Versions/3.10/lib/python3.10/genericpath.py", line 19, in exists
    os.stat(path)
TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType
  File "/Library/AutoPkg/autopkglib/__init__.py", line 873, in process
    self.env = processor.process()
stat: path should be string, bytes, os.PathLike or integer, not NoneType
Failed.
```

**Preferences contents**

```
vaughn@test-air ~ % defaults read ~/Library/Preferences/com.github.autopkg.plist
{
    "MUNKI_REPO" = "/Volumes/dm-munki";
    "RECIPE_REPOS" =     {
        "/Users/vaughn/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes" =         {
            URL = "https://github.com/autopkg/recipes";
        };
    };
    "RECIPE_SEARCH_DIRS" =     (
        ".",
        "~/Library/AutoPkg/Recipes",
        "/Library/AutoPkg/Recipes",
        "/Users/vaughn/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes"
    );
}
```

**AutoPkg output**
```
vaughn@test-air ~ % autopkg version
2.8.1
vaughn@test-air ~ % autopkg run -vvv munkitools6-signed.munki
Looking for munkitools6-signed.munki...
Found munkitools6-signed.munki in recipe map
**load_recipe time: 0.00169362499991621
Processing munkitools6-signed.munki...
WARNING: munkitools6-signed.munki is missing trust info and FAIL_RECIPES_WITHOUT_TRUST_INFO is not set. Proceeding...
{'AUTOPKG_VERSION': '2.8.1',
 'MUNKITOOLS_ADMIN_DESCRIPTION': 'Command-line Managed Software Center admin '
                                 'tools.',
 'MUNKITOOLS_ADMIN_DISPLAYNAME': 'Managed Software Center admin tools',
 'MUNKITOOLS_ADMIN_NAME': 'munkitools_admin',
 'MUNKITOOLS_APP_DESCRIPTION': 'Managed Software Center application.',
 'MUNKITOOLS_APP_DISPLAYNAME': 'Managed Software Center',
 'MUNKITOOLS_APP_NAME': 'munkitools',
 'MUNKITOOLS_APP_USAGE_DESCRIPTION': 'Application usage statistics for Managed '
                                     'Software Center.',
 'MUNKITOOLS_APP_USAGE_DISPLAYNAME': 'Managed Software Center app usage',
 'MUNKITOOLS_APP_USAGE_NAME': 'munkitools_app_usage',
 'MUNKITOOLS_CORE_DESCRIPTION': 'Core command-line tools used by Managed '
                                'Software Center.',
 'MUNKITOOLS_CORE_DISPLAYNAME': 'Managed Software Center core tools',
 'MUNKITOOLS_CORE_NAME': 'munkitools_core',
 'MUNKITOOLS_LAUNCHD_DESCRIPTION': 'launchd configuration files for use by '
                                   'Managed Software Center.',
 'MUNKITOOLS_LAUNCHD_DISPLAYNAME': 'Managed Software Center launchd files',
 'MUNKITOOLS_LAUNCHD_NAME': 'munkitools_launchd',
 'MUNKITOOLS_PYTHON_DESCRIPTION': 'Embedded Python tools used by Managed '
                                  'Software Center.',
 'MUNKITOOLS_PYTHON_DISPLAYNAME': 'Managed Software Center embedded python',
 'MUNKITOOLS_PYTHON_NAME': 'munkitools_python',
 'MUNKI_CATALOG': 'development',
 'MUNKI_CATEGORY': 'munkitools',
 'MUNKI_DEVELOPER': 'The Munki Project',
 'MUNKI_ICON': '',
 'MUNKI_REPO': '/Volumes/dm-munki',
 'MUNKI_REPO_SUBDIR': 'munkitools',
 'NAME': 'munkitools6',
 'PARENT_RECIPES': [],
 'RECIPE_CACHE_DIR': '/Users/vaughn/Library/AutoPkg/Cache/com.github.autopkg.munki.munkitools6-signed',
 'RECIPE_DIR': '/Users/vaughn/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes/munkitools',
 'RECIPE_OVERRIDE_DIRS': ['/Users/vaughn/Library/AutoPkg/RecipeOverrides'],
 'RECIPE_PATH': '/Users/vaughn/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes/munkitools/munkitools6-signed.munki.recipe',
 'RECIPE_REPOS': {'/Users/vaughn/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes': {'URL': 'https://github.com/autopkg/recipes'}},
 'RECIPE_SEARCH_DIRS': ['.',
                        '~/Library/AutoPkg/Recipes',
                        '/Library/AutoPkg/Recipes',
                        '/Users/vaughn/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes'],
 'verbose': 3}
GitHubReleasesInfoProvider
{'Input': {'asset_regex': '^munkitools-6.*?pkg$',
           'github_repo': 'macadmins/munki-builds'}}
GitHubReleasesInfoProvider: No value supplied for CURL_PATH, setting default value of: /usr/bin/curl
GitHubReleasesInfoProvider: No value supplied for GITHUB_URL, setting default value of: https://api.github.com
GitHubReleasesInfoProvider: No value supplied for GITHUB_TOKEN_PATH, setting default value of: ~/.autopkg_gh_token
Traceback (most recent call last):
  File "/Library/AutoPkg/autopkglib/__init__.py", line 873, in process
    self.env = processor.process()
  File "/Library/AutoPkg/autopkglib/__init__.py", line 656, in process
    self.main()
  File "/Library/AutoPkg/autopkglib/GitHubReleasesInfoProvider.py", line 202, in main
    releases = self.get_releases(
  File "/Library/AutoPkg/autopkglib/GitHubReleasesInfoProvider.py", line 132, in get_releases
    github = autopkglib.github.GitHubSession(
  File "/Library/AutoPkg/autopkglib/github/__init__.py", line 56, in __init__
    self.token = self._get_token(token_path=token_abspath)
  File "/Library/AutoPkg/autopkglib/github/__init__.py", line 64, in _get_token
    if not os.path.exists(pref_token_path) and os.path.exists(token_path):
  File "/Library/AutoPkg/Python3/Python.framework/Versions/3.10/lib/python3.10/genericpath.py", line 19, in exists
    os.stat(path)
TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType
  File "/Library/AutoPkg/autopkglib/__init__.py", line 873, in process
    self.env = processor.process()
stat: path should be string, bytes, os.PathLike or integer, not NoneType
Failed.
Receipt written to /Users/vaughn/Library/AutoPkg/Cache/com.github.autopkg.munki.munkitools6-signed/receipts/munkitools6-signed-receipt-20230816-084806.plist

The following recipes failed:
    munkitools6-signed.munki
        Error in com.github.autopkg.munki.munkitools6-signed: Processor: GitHubReleasesInfoProvider: Error: stat: path should be string, bytes, os.PathLike or integer, not NoneType

Nothing downloaded, packaged or imported.
vaughn@test-air ~ % 
```

**Expected behavior**
munkitools6-signed.munki works correctly under AutoPkg 2.7.2

**Version (please complete the following information):**
 - OS version: 13.5
- AutoPkg Version: 2.8.1RC6

**Comment by jgstew** [comment]
Switching to v2.8.1RC5 fixed the issue for me, so it has to be something introduced between v2.8.1RC5 and v2.8.1RC6

**Comment by nmcspadden** [comment]
@homebysix yep, let's just merge that in. When we merge this branch back into dev, we'll do another total audit of any hardcoded paths and replace them with constants.

**Comment by nmcspadden** [comment]
This is now resolved in 3.0.0 RC1.
